shared/domains/chelonia/internals.js

```diff
@@ -1293,53 +1296,69 @@ export default (sbp('sbp/selectors/register', {
       return
     }
 
-    // We check this.subscriptionSet to see if we're already
-    // subscribed to the contract; if not, we call sync.
-    if (!this.subscriptionSet.has(contractID)) {
-      await sbp('chelonia/private/in/syncContract', contractID)
-    }
+    try {
+      // Since we require a contract state for the duration of this operation,
+      // we call retain on this contract ephemerally and release it when we're
+      // done. This prevent the contract form being removed while this function
+      // is executing. We need to do reference counting manually here because
+      // we can't block on the contract queue
```

Comment:
If release operations are done on the same queue as this selector, then we wouldn't need to do this. Can you check if this is the case? If not, can you update the comment to explain why it's not the case?